lib/SIL/ValueOwnership.cpp

```diff
@@ -259,7 +254,7 @@ ValueOwnershipKindClassifier::visitForwardingInst(SILInstruction *I,
 #define FORWARDING_OWNERSHIP_INST(INST)                                        \
   ValueOwnershipKind ValueOwnershipKindClassifier::visit##INST##Inst(          \
       INST##Inst *I) {                                                         \
-    return visitForwardingInst(I);                                             \
+    return I->getOwnershipKind();                                              \
```

Comment:
This is confusing because `getOwnershipKind` is implemented with `ValueOwnershipKindClassifier`, so it looks like circular recursion. If you're relying on method overriding, then that should be commented.

Reply:
I am not. getOwnershipKind is on SILValue. This is calling on a SILInstruction.